### PR TITLE
Fix linter warnings

### PR DESCRIPTION
**Pull request by ericgaspar**
- Fix linter warnings
- New permissions
- Cleaning up

## PR Status
- [ ] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [ ] Can be reviewed and tested.

## Package_check results
---
*If you have access to [App Continuous Integration for packagers](https://yunohost.org/#/packaging_apps_ci) you can provide a link to the package_check results like below, replacing '-NUM-' in this link by the PR number and USERNAME by your username on the ci-apps-dev. Or you provide a screenshot or a pastebin of the results*

[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/writefreely_ynh%20PR80%20(ericgaspar)/)

### CODE REVIEW

**Review by anmol26s — APPROVED** [review]

File: scripts/_common.sh [hunk 1]
```diff
@@ -15,6 +15,19 @@
...
+  local USER=$1
+  shift 1
+
+  if [[ $USER = $(whoami) ]]; then
+    eval "$@"
+  else
+    sudo -u "$USER" "$@"
+  fi
+}
+
```
Comment (yalh76) [inline scripts/_common.sh:30]: Sorry but I don't want to use that helper and that version of this helper

File: scripts/backup [hunk 2]
```diff
@@ -6,7 +6,6 @@
 # IMPORT GENERIC HELPERS
 #=================================================
 
-#Keep this path for calling _common.sh inside the execution's context of backup and restore scripts
```
Comment (yalh76) [inline scripts/backup:9]: I usually keep that comment as a warning for anyone wanted to work on the package

File: scripts/change_url [hunk 3]
```diff
@@ -29,15 +29,10 @@ ynh_script_progression --message="Loading installation settings..."
 # Needed for helper "ynh_add_nginx_config"
 final_path=$(ynh_app_setting_get --app=$app --key=final_path)
 
-# Add settings here as needed by your application
-#db_name=$(ynh_app_setting_get --app=$app --key=db_name)
-#db_user=$db_name
-#db_pwd=$(ynh_app_setting_get --app=$app --key=db_pwd)
-
```
Comment (yalh76) [inline scripts/change_url:36]: Also keep those comment for anycase

File: scripts/install [hunk 4]
```diff
@@ -153,12 +153,12 @@ ynh_store_file_checksum --file="$final_path/config.ini"
...
-chown -R "$app":"$app" "$final_path"
+chown -R $app: $final_path
 
 pushd $final_path
-	sudo -u $app ./writefreely --init-db
-	sudo -u $app ./writefreely --gen-keys
-	sudo -u $app ./writefreely --create-admin "$admin":"$password"
+	ynh_exec_as $app ./writefreely --init-db
+	ynh_exec_as $app ./writefreely --gen-keys
+	ynh_exec_as $app ./writefreely --create-admin "$admin":"$password"
```
Comment (yalh76) [inline scripts/install:161]: I don't want to use that helper

File: scripts/install [hunk 5]
```diff
@@ -194,14 +194,15 @@ ynh_script_progression --message="Configuring SSOwat..."
 # Make app public if necessary
 if [ $is_public -eq 1 ]
 then
-	# unprotected_uris allows SSO credentials to be passed anyway.
-	ynh_app_setting_set --app=$app --key=unprotected_uris --value="/"
+	# Everyone can access the app.
+	# The "main" permission is automatically created before the install script.
+	ynh_permission_update --permission "main" --add "visitors"
```
Comment (yalh76) [inline scripts/install:199]: Waiting for yunohost 4.1 and permission system improved

File: scripts/restore [hunk 6]
```diff
@@ -6,7 +6,6 @@
 # IMPORT GENERIC HELPERS
 #=================================================
 
-#Keep this path for calling _common.sh inside the execution's context of backup and restore scripts
```
Comment (yalh76) [inline scripts/restore:9]: I prefer to keep this line as a warning

File: scripts/restore [hunk 7]
```diff
@@ -92,7 +91,7 @@ ynh_mysql_connect_as --user=$db_user --password=$db_pwd --database=$db_name < ./
 ynh_script_progression --message="Restoring the systemd configuration..."
 
 ynh_restore_file --origin_path="/etc/systemd/system/$app.service"
-systemctl enable $app.service
+systemctl enable $app.service --quiet
```
Comment (yalh76) [inline scripts/restore:94]: this should be added in example_ynh

File: scripts/upgrade [hunk 8]
```diff
@@ -72,14 +71,6 @@ ynh_app_setting_delete --app=$app --key=public_stats
...
 ynh_app_setting_delete --app=$app --key=local_timeline
 
-# Close a port
-
-if yunohost firewall list | grep -q "\- $port$"
-then
-	echo "Close port $port" >&2
-	yunohost firewall disallow TCP $port 2>&1
-fi
-
```
Comment (yalh76) [inline scripts/upgrade:82]: Those lines need to be kept because old version of the package opened the port ...

File: scripts/upgrade [hunk 9]
```diff
@@ -139,12 +130,12 @@ ynh_system_user_create --username=$app --home_dir="$final_path"
...
 
-chown -R "$app":"$app" "$final_path"
+chown -R $app: $final_path
 
 pushd $final_path
 	ynh_mysql_execute_as_root --sql="ALTER DATABASE $db_name CHARACTER SET latin1 COLLATE latin1_swedish_ci;" --database=$db_name
-	sudo -u $app ./writefreely db migrate
-	sudo -u $app ./writefreely --migrate
+	ynh_exec_as $app ./writefreely db migrate
+	ynh_exec_as $app ./writefreely --migrate
```
Comment (yalh76) [inline scripts/upgrade:138]: still don't want to use that helper

File: scripts/upgrade [hunk 10]
```diff
@@ -187,24 +178,12 @@ yunohost service add $app --description "$app daemon for WriteFreely" --log "/va
...
-# SETUP SSOWAT
-#=================================================
-ynh_script_progression --message="Upgrading SSOwat configuration..."
-
-# Make app public if necessary
-if [ $is_public -eq 1 ]
-then
-	# unprotected_uris allows SSO credentials to be passed anyway
-	ynh_app_setting_set --app=$app --key=unprotected_uris --value="/"
-fi
```
Comment (yalh76) [inline scripts/upgrade:202]: Waiting to new permission system

**Review by yalh76 — CHANGES_REQUESTED** [review]

**Comment by ericgaspar** [comment]
Closing as there not much left to keep in this PR

**Pull request closed without merging**